src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -15,22 +16,22 @@ internal class EnumerableRewriter : ExpressionVisitor
     {
         // We must ensure that if a LabelTarget is rewritten that it is always rewritten to the same new target
         // or otherwise expressions using it won't match correctly.
-        private Dictionary<LabelTarget, LabelTarget> _targetCache;
+        private Dictionary<LabelTarget, LabelTarget>? _targetCache;
         // Finding equivalent types can be relatively expensive, and hitting with the same types repeatedly is quite likely.
-        private Dictionary<Type, Type> _equivalentTypeCache;
+        private Dictionary<Type, Type>? _equivalentTypeCache;
 
         protected override Expression VisitMethodCall(MethodCallExpression m)
         {
-            Expression obj = Visit(m.Object);
+            Expression? obj = Visit(m.Object);
```

Comment:
Visit method annotated with [return NotNullIfNotNull("node")], as m.Object is not null here returning expression is not null. Then we will not need `!` below for the `obj`

Reply:
It's theoretically possible for `m.Object` to be null, for example if `m` is of runtime type `MethodCallExpression`, for which the property [always evaluates to `null`](https://github.com/dotnet/runtime/blob/2c54bd95ab18962fa6a2c51ef9871d8b6f57d91b/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs#L25). Even if it is the case that we only consume proper subtypes which are guaranteed to return non-null values, I think it would make not to let that invariant influence annotations in this context, particularly since it doesn't really impact public APIs.